api/v1beta1/azurecluster_validation.go

```diff
@@ -368,6 +353,10 @@ func validateAPIServerLB(lb LoadBalancerSpec, old LoadBalancerSpec, cidrs []stri
 			}
 		}
 
+		// FIXME: Is this validation really dependent on the the validation of FrontEndIPs?
```

Comment:
We can move this to `validateClassSpecForAPIServerLB`. It has no dependency on frontend ip count